src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,329 @@
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using Iot.Device.Ssd1351.Command;
+
+namespace Iot.Device.Ssd1351
+{
+    /// <summary>
+    /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+    /// light emitting diode dot-matrix graphic display system. 
+    /// </summary>
+    public partial class Ssd1351 : IDisposable
+    {
+
+        private readonly SpiDevice _spiDevice;
+        private readonly GpioController _gpioDevice;
+        private readonly int _dcPinId;
+        private readonly int _resetPinId;
+        private ColorDepth _colorDepth;
+        private ColorSequence _colorSequence;
+        private int _spiBufferSize;
+
+        private const byte SCREEN_WIDTH_PX = 128;
+        private const byte SCREEN_HEIGHT_PX = 128;
+        private const int DefaultSPIBufferSize = 0x1000;
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">>The SPI device used for communication.</param>
+        /// <param name="gpioDevice">>The GPIO controller device used for communication.</param>
+        /// <param name="dcPinId">>The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPinId">>The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        public Ssd1351(SpiDevice spiDevice, GpioController gpioDevice, int dcPinId, int resetPinId)
+        {
+            _spiDevice = spiDevice;
+            _gpioDevice = gpioDevice;
+            _dcPinId = dcPinId;
+            _resetPinId = resetPinId;
+
+            _gpioDevice.OpenPin(_dcPinId, PinMode.Output);
+            _gpioDevice.OpenPin(_resetPinId, PinMode.Output);
+
+            SPIBufferSize = DefaultSPIBufferSize;
+
+            SendCommand(new SetSegmentReMapColorDepth());
+
+        }
+
+        /// <summary>
+        /// The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions
+        /// </summary>
+        public int SPIBufferSize
+        {
+            get => _spiBufferSize;
+            set
+            {
+                if(!InRange((uint)value, 0x1000, 0x10000))
+                {
+                    throw new ArgumentException($"SPI Buffer Size must be between 4096 and 65536.", nameof(SPIBufferSize));
+                }
+
+                _spiBufferSize = value;
+            }
+        }
+
+        /// <summary>
+        /// Convert a color structure to a byte tuple represening the colour in 565 format.
+        /// </summary>
+        /// <param name="color">The color to be converted.</param>
+        private (byte, byte) Color565(Color color)
```

Comment:
@krwq didn't we have this already somewhere in the repo? I kind of feel that we already did this and had a common library for this but I can't seem to find it. I think we did it for sense hat, but I might be wrong.

Reply:
I think there was a discussion regarding this that I saw. I'll have a look around. It is a bit more complex than first it seems though as with the Ssd1351 you can change the byte order of the colour representation and this is also handled within the method.

Reply:
I had a look and couldn't find reference to this right now.

Reply:
I have found the method doing an RGB565 conversion. It was in the WriteColor method implemented in SenseHatLedMatrixSysFs.cs It doesn't look like this has been broken out itno a common library and is not really a generic function but one that is tied to writing to the "RPi-Sense FB" device and so is not very generic in its current form.

Reply:
Yes, we might want to move this to common folder:
https://github.com/dotnet/iot/blob/master/src/devices/SenseHat/SenseHatLedMatrixSysFs.cs#L95

you can leave as is for now though - it might be worth creating an issue to unify this and have some conversation how to make it generic purpose (i.e. create Color565 struct or just represent the value as i.e. ushort or tuple as in here).